saleor/static/dashboard-next/components/AssignCollectionDialog/AssignCollectionDialog.tsx

```diff
@@ -0,0 +1,176 @@
+      maxWidth="sm"
+    >
+      <Form initial={initialForm} onSubmit={onSubmit}>
+        {({ data, change }) => (
+          <>
+            <DialogTitle>{i18n.t("Assign Collection")}</DialogTitle>
+            <DialogContent className={classes.overflow}>
+              <Debounce debounceFn={onFetch}>
+                {fetch => (
+                  <TextField
+                    name="query"
+                    value={data.query}
+                    onChange={event => change(event, () => fetch(data.query))}
+                    label={i18n.t("Search Collection", {
+                      context: "product search input label"
+                    })}
+                    placeholder={i18n.t(
+                      "Search by product name, attribute, product type etc...",
+                      {
+                        context: "product search input placeholder"
+                      }
+                    )}
+                    fullWidth
+                    InputProps={{
+                      autoComplete: "off",
+                      endAdornment: loading && <CircularProgress size={16} />
+                    }}
+                  />
+                )}
+              </Debounce>
+              <FormSpacer />
+              <Table>
+                <TableBody>
+                  {collections &&
+                    collections.map(category => (
+                      <TableRow key={category.id}>
+                        <TableCell
+                          padding="checkbox"
+                          className={classes.checkboxCell}
+                        >
+                          <Checkbox
+                            checked={
+                              !!data.collections.find(
+                                selectedCollection =>
+                                  selectedCollection.id === category.id
+                              )
+                            }
+                            onChange={() =>
+                              data.collections.find(
+                                selectedCollection =>
+                                  selectedCollection.id === category.id
+                              )
+                                ? change({
+                                    target: {
+                                      name: "collections",
+                                      value: data.collections.filter(
+                                        selectedCollection =>
+                                          selectedCollection.id !== category.id
+                                      )
+                                    }
+                                  } as any)
+                                : change({
+                                    target: {
+                                      name: "collections",
+                                      value: [...data.collections, category]
+                                    }
+                                  } as any)
```

Comment:
Because current form implementation is retarded and operates on events, which can't be easily synthesized. So I just give it minimal data and type it as `any`, so TS won't bother me with synthesizing useless DOM events.